src/app/qgsidentifyresultsdialog.cpp

```diff
@@ -1395,7 +1395,7 @@ void QgsIdentifyResultsDialog::addFeature( QgsPointCloudLayer *layer, const QStr
     feature.setAttribute( fieldIndex, v );
   }
 
-  const QgsGeometry selectionGeometry( QgsGeometry::fromPointXY( QgsPointXY( attributes[QStringLiteral( "X" )].toDouble(), attributes[QStringLiteral( "Y" )].toDouble() ) ) );
+  const QgsGeometry selectionGeometry( QgsGeometry::fromPoint( QgsPoint( attributes[QStringLiteral( "X" )].toDouble(), attributes[QStringLiteral( "Y" )].toDouble(), attributes[QStringLiteral( "Z" )].toDouble() ) ) );
```

Comment:
Does this work when offset/scale is applied to point cloud in layer properties? In such case, the "Z" attribute should be replaced by "Z (original)" and "Z (adjusted)" (translatable strings)

Reply:
Ah, good catch, it's a bit ugly to rely on a translatable string here...
I fixed this, but can't think of an easy way to update existing rubberbands when the layer's scale/offset changes.